core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -87,9 +100,8 @@
  * @author Mircea Markus
  * @since 6.0
  */
-@Store
 @ConfiguredBy(SingleFileStoreConfiguration.class)
-public class SingleFileStore<K, V> implements AdvancedLoadWriteStore<K, V> {
+public class SingleFileStore<K, V> implements NonBlockingStore<K, V> {
    private static final Log log = LogFactory.getLog(SingleFileStore.class);
 
    public static final byte[] MAGIC_BEFORE_11 = new byte[]{'F', 'C', 'S', '1'}; //<11
```

Comment:
Shouldn't we add a new MAGIC number for 13 since it migrates data?

Reply:
I thought about it, but then it's exactly the same format as before. 

The data is migrated when the store is segmented, but I don't see a benefit in changing the magic number there either: when you start an older version of Infinispan after the data was migrated to a single file, the old version won't read the migrated file at all, so it won't see the new magic number.

Reply:
Doesn't the `migrateNonSegmented` always get invoked though or how does it verify this was completed?

Reply:
If the file exists in the right place and has the latest magic header, `validateExistingFile()` returns `null` and there is no migration.

Reply:
Ah sorry I misread `if (!Arrays.equals(MAGIC_LATEST, headerMagic))`